lib/Onnxifi/Base.cpp

```diff
@@ -232,7 +232,24 @@ onnxStatus Graph::setIOAndRun(uint32_t inputsCount,
       ONNX_NAMESPACE::GraphProto inputG;
       for (const auto &p : ctx->getPlaceholderBindings()->pairs()) {
         auto *t = inputG.add_initializer();
-        ONNXModelWriter::writeTensor(*p.second, t);
+        const auto &inputTensor = *p.second;
+        size_t unpaddedSize = inputTensor.getUnpaddedSizeInBytes();
+        size_t tensorSize = inputTensor.getSizeInBytes();
+        if (unpaddedSize == tensorSize) {
+          ONNXModelWriter::writeTensor(inputTensor, t);
+        } else {
+          // If the input is a partial tensor, then save only the part that has
+          // data.
+          auto ty = inputTensor.getType();
+          auto dims = ty.dims().vec();
+          dims[0] = dims[0] / (tensorSize / unpaddedSize);
```

Comment:
I don't understand the logic here -- I think we want to do something like `dims[0] = unpaddedSize / ty.getElementSize()`, no?

Reply:
dims[0] = unpaddedSize / ty.getElementSize() would only work if the tensor is 1D. From my testing on real models, partial tensor doesn't have to be 1D. So say we have a 8x10 tensor and it's partially populated with 2 entries in it, then we just want to take a slice of 2. So dim 0 needs to be (8 / 4). We get 4 from (tensorSize / unpaddedSize) = (8*10*element_size) / (2*10*element_size) = 4. 

So altogether it's dims[0] / (tensorSize / unpaddedSize).